mediaflow_proxy/routes/extractor.py

```
@@ -0,0 +1,32 @@
+        extractor = ExtractorFactory.get_extractor(extractor_params.host, extractor_params.use_request_proxy)
+        final_url, headers = await extractor.extract(extractor_params.destination)
+
+        if extractor_params.redirect_stream:
+            formatted_headers = "&".join(f"h_{k}={v}" for k, v in headers.items())
+            stream_url = f"/proxy/stream?api_password={settings.api_password}&d={final_url}&{formatted_headers}"
+            return RedirectResponse(url=stream_url)
+
+        return {"url": final_url, "headers": headers}
+
+    except ValueError as e:
+        raise HTTPException(status_code=400, detail=str(e))
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)